### PR TITLE
implement "delete a commons" functionality in `commonsController.java`

### LINKED ISSUE #32: 6pm-4: Add DELETE method to the commons controller.
- [x] There is a DELETE method in the `Commons` controller that accepts the identifier of a `Commons` as a single query parameter and deletes the `Commons` from the backend.
- [x] the delete method has full test coverage
- [x] the delete method has full mutation coverage

### PR DESCRIPTION
**Pull request by rsdharmaji**
# Overview

the delete method is added to the commons controller

# Issues Addressed

closes #32 

# Details

delete method supports an id for deletion, and throws an `EntityNotFoundException` if deleting nonexistent commons.
`ADMIN` authentication is required to use the delete method.

here is an "after" image:
image

### CODE REVIEW

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java [hunk 1]
```diff
@@ -154,4 +156,54 @@ public void joinCommonsTest() throws Exception {
...
+              delete("/api/commons?id=2")
+                      .with(csrf()))
+              .andExpect(status().is(404)).andReturn();
+      
+      verify(commonsRepository, times(1)).findById(2L);
+      
+
+      String responseString = response.getResponse().getContentAsString();
+      
+      String expectedString = "{\"message\":\"Commons with id 2 not found\",\"type\":\"EntityNotFoundException\"}";
```
Comment (jacksoncooper) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:409]: If we're okay not implementing [HTTP DELETE to spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE), this looks good. For now, let's merge and focus on getting the commons list built.

**Review by jacksoncooper — DISMISSED** [review]
Use proper HTTP statuses when the server replies later.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java [hunk 2]
```diff
@@ -125,6 +125,19 @@ public Commons getCommonsById(
...
+  @ApiOperation(value = "Delete a Commons")
+  @PreAuthorize("hasRole('ROLE_ADMIN')")
+  @DeleteMapping("")
+  public ResponseEntity<String> deleteCommons(
+          @ApiParam("id") @RequestParam Long id) {
+      
+      Commons foundCommons = commonsRepository.findById(id).orElseThrow( ()->new EntityNotFoundException(Commons.class, id));
+ 
+      commonsRepository.deleteById(id);
+      return ResponseEntity.ok().body(String.format("commons with id %d deleted", id));
```
Comment (btk5h) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java]: API controller methods should ideally return JSON responses at all times. There is a `ApiController#genericMessage` method to help you generate that. If you return the object generated by that method in your controller, Spring Boot will format it correctly as JSON.

**Review by btk5h — DISMISSED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java (hunk 1 above)
Comment (rsdharmaji) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:409]: i think its fine for now

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java (hunk 2 above)
Comment (rsdharmaji) [reply src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java]: solved in latest commit.

**Review by Jacqueskim — DISMISSED** [review]
LGTM

**Comment by kheff16** [comment]
Needs rebase or merge with main but otherwise lgtm

**Review by jacksoncooper — DISMISSED** [review]
I don't think we broke anything merging.

**Comment by jacksoncooper** [comment]
#52 affects this.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java [hunk 3]
```diff
@@ -124,6 +126,23 @@ public Commons getCommonsById(
...
+  @ApiOperation(value = "Delete a Commons")
+  @PreAuthorize("hasRole('ROLE_ADMIN')")
+  @DeleteMapping("")
+  public Object deleteCommons(
+          @ApiParam("id") @RequestParam Long id) {
+      
+      Commons foundCommons = commonsRepository.findById(id).orElseThrow( ()->new EntityNotFoundException(Commons.class, id));
+ 
+      commonsRepository.deleteById(id);
+      userCommonsRepository.deleteAllByCommonsId(id);
```
Comment (jacksoncooper) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java:168]: This destroys both the `COMMONS` and `USER_COMMONS` tables.

**Review by jacksoncooper — DISMISSED** [review]
This patch doesn't work, unfortunately.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java (hunk 3 above)
Comment (rsdharmaji) [reply src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java:168]: no way to fix this with the given setup it seems.

**Pull request merged**